### ISSUE TITLE
Weird Dragging Behavior when dragging child nodes

### ISSUE THREAD

**Issue by dthian** [viromedia/virocore#109]
We see weird dragging behavior when attempting to drag child nodes within parent objects that are rotated (the items are dragged on an arch, instead of in the direction of the drag).

**Comment by rgipd** [comment]
@dthian I don't know if it's related to this bug, but how can I fix a z position in the drag listener? This is possible or I can't do it because of this drag bug?


My code:
```
node.setDragListener { source, node, worldLocation, localLocation ->
                localLocation.z = 0.05f / 2.0f; // FIXED Z

                val v = Vector(container!!.node.positionRealtime.x, container!!.node.positionRealtime.y, 0.05f/2.0f); // FIXED Z
                node.setPosition(v);
            }
```

About the scene:
  The node is a child of a big box node (that is child of the root of the scene).
  The camera is in orbit (ViroViewScene custom pan to orbit, like the bug #30 you put the source code) and the view is "rotated" by the orbit camera.
  No nodes is rotated.

My objective:
  I just want to be able to drag the node and, even with the orbit camera showing 45 deg around the main big box) I don't want to the drag affect the Z position of my node. I want it to be fixed (my custom value or it's own value).

 In iOS Scene Kit, I could do it with Pan Gesture. I can do this in Android as well with TouchListener, but I couldn't find any hitTest function in ViroViewScene. So I'm using the DragListener code above. It works, but I really want to make the z-position fixed somehow.


Thanks.

**Comment by rgipd** [comment]
@dthian So, I do a clickListener in Object3d Node, then I handle by ViroViewScene touchListener for the drag, can be a good idea.

I think the feature request would help if you, could choose the axis before the drag (or choose a fixed value to some axis before the drag) I think it could be ideal.
Maybe with mask? fixedDragMask = DRAG_X | DRAG_Y;
Something like that?

At least, a hitTest method could be useful as well :)

**Comment by dthian** [comment]
Thanks @rgipd, that's a great idea - I've added your mask suggestion to the feature request.

With regards to a virtual hit test - what did you have in mind again to use this API with, that a click event with provided world coordinates would not provide? Thanks!

**Comment by dthian** [comment]
This has been fixed in the latest release of ViroCore 1.7.2. Please checkout the update instructions here: https://virocore.viromedia.com/docs/releases
